microservices/matchmaking-service/src/scrim/scrim.service.spec.ts

```diff
@@ -0,0 +1,625 @@
+    });
+
+    describe("Creating a Scrim", () => {
+        it("Should throw with PlayerAlreadyInScrim when a player tries to join a scrim while in another scrim", async () => {
+            const createScrimData: CreateScrimOptions = {
+                ...scrimIds(),
+                settings: scrimSettings(3, 2, ScrimMode.ROUND_ROBIN, true, false, 1000),
+                join: {
+                    playerId: 1,
+                    playerName: "HyperCoder",
+                    leaveAfter: 1000,
+                },
+            };
+    
+            playerInAnyScrim.mockImplementationOnce(async () => true);
+
+            const actual = async (): Promise<Scrim> => service.createScrim(createScrimData);
+            const expected = MatchmakingError.PlayerAlreadyInScrim;
+
+            await expect(actual).rejects.toThrow(expected);
+        });
+    
+        it("Should throw with ScimGroupNotSupported when a player tries to create a round robin scrim with groups", async () => {
+            const createScrimData: CreateScrimOptions = {
+                ...scrimIds(),
+                settings: scrimSettings(3, 2, ScrimMode.ROUND_ROBIN, true, false, 1000),
+                join: {
+                    playerId: 1,
+                    playerName: "HyperCoder",
+                    leaveAfter: 1000,
+                    createGroup: true,
+                },
+            };
+    
+            playerInAnyScrim.mockImplementationOnce(async () => false);
+
+            const actual = async (): Promise<Scrim> => service.createScrim(createScrimData);
+            const expected = MatchmakingError.ScrimGroupNotSupportedInMode;
+
+            await expect(actual).rejects.toThrow(expected);
+        });
+    
+        it("Should create a Round Robin scrim successfully", async () => {
+            const createScrimData: CreateScrimOptions = {
+                ...scrimIds(),
+                settings: scrimSettings(3, 2, ScrimMode.ROUND_ROBIN, true, false, 1000),
+                join: {
+                    playerId: 1,
+                    playerName: "HyperCoder",
+                    leaveAfter: 1000,
+                },
+            };
+            const scrim: Scrim = {
+                id: "hello world!",
+                createdAt: someDate,
+                updatedAt: someDate,
+                status: ScrimStatus.PENDING,
+                ...scrimIds(),
+                players: [
+                    players.hyper(someDate),
+                ],
+                games: undefined,
+                settings: createScrimData.settings,
+            };
+            
+            playerInAnyScrim.mockImplementationOnce(async () => false);
+            createScrim.mockImplementationOnce(async () => scrim);
+            updatePlayer.mockImplementationOnce(async () => {});
+
+            const actual = await service.createScrim(createScrimData);
+            const expected: Scrim = scrim;
```

Comment:
Instead of renaming `scrim` to `expected` here, just call it `expected` on line `135`